libsolidity/interface/CompilerStack.h

```diff
@@ -367,7 +373,8 @@ class CompilerStack: public langutil::CharStreamProvider
 		std::string yulIROptimized; ///< Optimized experimental Yul IR code.
 		std::string ewasm; ///< Experimental Ewasm text representation
 		evmasm::LinkerObject ewasmObject; ///< Experimental Ewasm code
-		util::LazyInit<std::string const> metadata; ///< The metadata json that will be hashed into the chain.
+		util::LazyInit<std::string const> evmMetadata; ///< The metadata json that will be hashed into the chain. For EVM codegen.
+		util::LazyInit<std::string const> irMetadata; ///< The metadata json that will be hashed into the chain. For IR codegen.
```

Comment:
Why do we need two different metadatas here, but only one `evmAssembly`?

Reply:
Because we only produce one assembly but we might need to print two different metadata hashes. Each metadata json is appropriate for the bytecode from a different codegen but the assembly can only be generated by one of them.

If you prefer I could store only one metadata (matching `viaIR` is it was until this PR) and generate metadata for IR on the fly and discard it immediately after getting the hash for IR codegen.

Reply:
I think it is less confusing to only store the metadata json that is printed for `--metadata` - unless it is a big performance drawback. And even if it is, it only will be for `--experimental-via-ir` or `--ir`, right?

Reply:
ok. Either is fine with me. I thought that we'd want to store it to be able to print it later without generating it again but if we don't need that then generating it on the fly should be fine.

Reply:
Done.